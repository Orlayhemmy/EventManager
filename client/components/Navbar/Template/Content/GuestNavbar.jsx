import React from 'react';
import { Link } from 'react-router-dom';
import SignIn from '../Form/SignIn';

export default class Navbar extends React.Component {
     /**
   * @memberof NavBar
   * @method render
   * @description it renders the content of guestlinks
   * @returns the HTML of NavBar
   */
  render() {
    const { navbarProps } = this.props;
    return (
      <ul className="navbar-nav ml-auto">
        <li className="nav-item">
          <Link to="/" className="nav-link">
            <span className="nav-link-text">Home</span>
          </Link>
        </li>
        <li className="nav-item">
          <Link to="/view-centers" className="nav-link">
            <span className="nav-link-text">centers</span>
          </Link>
        </li>
        <li className="nav-item">
          <Link to="/about" className="nav-link">
            <span className="nav-link-text">about us</span>
          </Link>
        </li>
        <li className="nav-item">
          <div className="dropdown">
            <button
              class="btn btn-success dropdown-toggle main-color-bg"
              data-toggle="dropdown"
              aria-haspopup="true"
              aria-expanded="false"
              type="button"
              id="signIn"
            >
              Sign In
            </button>
            <div
              class="dropdown-menu dropdown-menu-right"
              aria-labelledby="signIn"
            >
              <SignIn navbarProps={navbarProps}/>
            </div>
          </div>
        </li>
      </ul>
    );
  }
}