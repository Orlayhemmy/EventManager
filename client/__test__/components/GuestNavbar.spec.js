import React from 'react';
import GuestNavbar from '../../components/Navbar/Template/Content/GuestNavbar';
import { navbarProps } from './MockData';

describe('guest navbar', () => {
  const wrapper = shallow(<GuestNavbar navbarProps={navbarProps} />);
  it('should render the component without error', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
