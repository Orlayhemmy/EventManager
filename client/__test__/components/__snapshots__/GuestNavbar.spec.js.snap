// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`guest navbar should render the component based 1`] = `
<ul
  className="navbar-nav ml-auto"
>
  <li
    className="nav-item"
  >
    <Link
      className="nav-link active"
      replace={false}
      to="/"
    >
      <span
        className="nav-link-text"
      >
        Home
      </span>
    </Link>
  </li>
  <li
    className="nav-item"
  >
    <Link
      className="nav-link"
      replace={false}
      to="/view-centers"
    >
      <span
        className="nav-link-text"
      >
        centers
      </span>
    </Link>
  </li>
  <li
    className="nav-item"
  >
    <Link
      className="nav-link"
      replace={false}
      to="/about"
    >
      <span
        className="nav-link-text"
      >
        about us
      </span>
    </Link>
  </li>
  <li
    className="nav-item"
  >
    <div
      className="dropdown"
    >
      <button
        aria-expanded="false"
        aria-haspopup="true"
        className="btn btn-success dropdown-toggle main-color-bg"
        data-toggle="dropdown"
        id="signIn"
        type="button"
      >
        Sign In
      </button>
      <div
        aria-labelledby="signIn"
        className="dropdown-menu dropdown-menu-right mt-3"
      >
        <SignIn
          navbarProps={
            Object {
              "logout": [MockFunction],
              "path": "/",
            }
          }
        />
      </div>
    </div>
  </li>
</ul>
`;
